### ISSUE TITLE
# and proceeding characters being stripped from filenames during download

### ISSUE THREAD

**Issue by walkintom** [munki/munki#700]
Experienced on 10.11.6
Behaviour experienced with a .dmg file but may affect .pkg too.

ClientSoftware#company.dmg was downloaded by munki as just ClientSoftware, stripping #company.dmg from the filename, therefore the disk image couldn't be mounted.

**Comment by bochoven** [comment]
I don't think you could download that dmg with a browser.

**Comment by walkintom** [comment]
Fair point, and I've changed the filename but was still asked to log it as an issue anyway.

**Comment by gregneagle** [comment]
https://github.com/munki/munki/blob/master/code/client/munkilib/updatecheck/download.py#L130

(or in the 2.8.2 release code, https://github.com/munki/munki/blob/v2.8.2/code/client/munkilib/updatecheck.py#L822)

Should encode the '#' as '%23'.

Some testing and investigation needed. This might be an issue with NSURLSession/NSURLConnection, in which case we'll just have to document "Don't use the # character in anything that will be accessed via URL"
